examples/navier-stokes/navierstokes.c

```diff
@@ -291,24 +347,46 @@ static PetscErrorCode TSMonitor_NS(TS ts, PetscInt stepno, PetscReal time,
   // Print every 'outputfreq' steps
   if (stepno % user->outputfreq != 0)
     PetscFunctionReturn(0);
-
-  ierr = DMGetGlobalVector(user->dm, &U); CHKERRQ(ierr);
-  ierr = PetscObjectSetName((PetscObject)U, "StateVec");
-  ierr = VecScatterBegin(user->ltog, Q, user->Qloc, INSERT_VALUES,
-                         SCATTER_REVERSE); CHKERRQ(ierr);
-  ierr = VecScatterEnd(user->ltog, Q, user->Qloc, INSERT_VALUES,
-                       SCATTER_REVERSE); CHKERRQ(ierr);
-  ierr = MapLocalToDMGlobal(user, user->Qloc, U, scale); CHKERRQ(ierr);
+  ierr = DMGetLocalVector(user->dm, &Qloc);CHKERRQ(ierr);
+  ierr = PetscObjectSetName((PetscObject)Qloc, "StateVec"); CHKERRQ(ierr);
+  ierr = VecZeroEntries(Qloc);CHKERRQ(ierr);
+  ierr = DMGlobalToLocal(user->dm, Q, INSERT_VALUES, Qloc); CHKERRQ(ierr);
+  if (0) {
```

Comment:
Do we want to keep this?

Reply:
It's a longer term issue. I made this placeholder so I'll delete that dead code from the branch. https://gitlab.com/petsc/petsc/issues/564